File: src/DataTypes/MyPair.java
package DataTypes;

/**
 * Created by vladc on 07.11.2016.
 */
public class MyPair<F, S> {
    private F first;
    private S second;

    public MyPair(F first, S second) {
        this.first = first;
        this.second = second;
    }

    public F getFirst() {
        return first;
    }

    public S getSecond() {
        return second;
    }

    public void setFirst(F first) {
        this.first = first;
    }

    public void setSecond(S second) {
        this.second = second;
    }

    @Override
    public String toString() {
//        return "(" + first.toString() + ", " + second.toString() + ")";
        return first.toString();
    }

    @Override
    @SuppressWarnings("unchecked")
    public boolean equals(Object other) {
        if (other == null) return false;
        if (other == this) return true;
        if (!(other instanceof MyPair)) return false;

        MyPair<F, S> otherPair = (MyPair<F, S>)other;
        return otherPair.getFirst().equals(this.first) &&
                otherPair.getSecond().equals(this.second);
    }
}

File: Toy-Language-Interpreter/DataTypes/MyPair.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Toy_Language_Interpreter.DataTypes
{
    class MyPair<F, S>
    {
        private F first;
        private S second;

        public MyPair(F first, S second)
        {
            this.first = first;
            this.second = second;
        }

        public F GetFirst()
        {
            return first;
        }

        public S GetSecond()
        {
            return second;
        }

        public void SetFirst(F first)
        {
            this.first = first;
        }

        public void SetSecond(S second)
        {
            this.second = second;
        }

        public override string ToString()
        {
            return first.ToString();
        }
    }
}
